mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,261 @@
+# Author: Proloy Das <pdas6@mgh.harvard.edu>
+#
+# License: BSD-3-Clause
+import os
+import warnings
+import numpy as np
+import pytest
+
+from numpy.testing import assert_allclose
+
+from mne.io import read_raw_nsx
+from mne.io.nsx.nsx import _decode_online_filters
+from mne.io.meas_info import _empty_info
+from mne.datasets.testing import data_path, requires_testing_data
+from mne.io.tests.test_raw import _test_raw_reader
+from mne.io.constants import FIFF
+from mne import make_fixed_length_epochs
+
+
+testing_path = data_path(download=False)
+nsx_21_fname = os.path.join(testing_path, "nsx", "test_NEURALSG_raw.ns3")
```

Comment:
Excellent, I hope this helped adding example files, sorry if this was a bit of a pain to do all these versions but I think it will catch a lot of the issues in advance

Reply:
Not to be overly pedantic but is there another nsx file reader in matlab or anything that you could read these files in from and plot and check data values on? That would just be an extra sanity check since this file format is pretty complicated.

Reply:
yes, there is a primitive reader by Blackrock systems + another by neo IO, and I verified all files using them.